manifests/charts/ai-gateway-helm/templates/deployment.yaml

```diff
@@ -24,9 +24,9 @@ spec:
       labels:
         {{- include "ai-gateway-helm.controller.selectorLabels" . | nindent 8 }}
     spec:
-      {{- with .Values.controller.imagePullSecrets }}
+      {{- if include "ai-gateway-helm.controller.imagePullSecrets" . }}
```

Comment:
Here we are executing the helm template twice, when it's unnecessary... Probably we don't even need to include a helper and just use here the proposed default with the `with` statement>